File: api/src/domain/entities/usuario.py
from dataclasses import dataclass
from enum import Enum


class TipoUsuario(str, Enum):
    Admin = "Admin"
    Cajero = "Cajero"


@dataclass
class Usuario:
    UsuarioId: int | None
    NombreUsuario: str
    Contrasena: str
    TipoUsuario: TipoUsuario

File: web/src/types/usuario.ts
export type TipoUsuario = "Admin" | "Cajero";

export interface Usuario {
  UsuarioId: number;
  NombreUsuario: string;
  TipoUsuario: TipoUsuario;
}

export interface UsuarioCreate {
  NombreUsuario: string;
  Contrasena: string;
  TipoUsuario: TipoUsuario;
}

export interface UsuarioUpdate extends UsuarioCreate {}
